### ISSUE TITLE
white screen, cannot Burn Bootloader. 

### ISSUE THREAD

**Issue by keyboard-magpie** [CircuitMess/CircuitMess-Ringo#40]
Have managed to upload simple sketches which clears the white screen. 
Cannot upload standard firmware, errors as follows:

Arduino: 1.8.10 (Mac OS X), Board: "Ringo by CircuitMess, Minimal SPIFFS (Large APPS with OTA), 80MHz, 921600, None"

/Users/tombarnes/Library/Arduino15/packages/cm/tools/esptool_py/2.6.1/esptool --chip esp32 --port /dev/cu.SLAB_USBtoUART --baud 921600 --before default_reset --after hard_reset write_flash -z --flash_mode dio --flash_freq 80m --flash_size detect 0xe000 /Users/tombarnes/Library/Arduino15/packages/cm/hardware/esp32/1.0.2/tools/partitions/boot_app0.bin 0x1000 /Users/tombarnes/Library/Arduino15/packages/cm/hardware/esp32/1.0.2/tools/sdk/bin/bootloader_dio_80m.bin 0x10000 /Users/tombarnes/Library/Arduino15/packages/cm/hardware/esp32/1.0.2/firmware/firmware.bin 0x8000 /Users/tombarnes/Library/Arduino15/packages/cm/hardware/esp32/1.0.2/firmware/firmware.partitions.bin 
esptool.py v2.6
Serial port /dev/cu.SLAB_USBtoUART
Connecting........_____....._____....._____....._____....._____....._____.....____Error while burning bootloader.
_

A fatal error occurred: Failed to connect to ESP32: Invalid head of packet (0x73)"

**Comment by robbie8-bit** [comment]
Hi Tom,
It seems that your computer cannot connect to the ESP.
This sometimes can be a problem on Mac computers as they behave oddly in some cases.

First thing you can try to do is to switch the USB port then try again.

The other can be a power problem. On many laptops, the power provided on the USB port is not sufficient to power an ESP32 with activated WiFi and/or Bluetooth. The solution is to use a powered USB hub which can deliver 1A to the connected USB device.

Another issue is a missing driver. Depending on ESP32 module used it might be necessary to install a USB to UART bridge vcp driver. You can do that via [this link](https://www.silabs.com/products/development-tools/software/usb-to-uart-bridge-vcp-drivers).

The last thing to be done if nothing works is to install Developer SDK. You can check out how to do that [here](https://superuser.com/questions/781982/how-can-i-install-usb-prober-from-the-developer-sdk-on-mac-os-x). 

It can be as easy as just switching a port but then again it can be solved by installing some software, depending on the device. Try some of it out and contact us again.

Cheers,
Robert

**Comment by keyboard-magpie** [comment]
Fixed that; Bootloader now uploads. Am back to where I started though with a fixed white screen. Opening a serial monitor to the Ringo shows healthy output, other than SD card error in the first ten lines or so.

**Comment by robbie8-bit** [comment]
Okay, what did you write in Arduino code? Did you write for the program to clear the screen and set it to white?

Arduino sometimes automatically uploads your code even though you just click 'Burn bootloader'. Try to delete everything you've written and the default ringo software should upload just fine by itself.

**Comment by keyboard-magpie** [comment]
Nope. Blank sketch. fixed white screen.

**Comment by robbie8-bit** [comment]
I really have no idea why is your screen white.

If the bootloader goes through well and everything gets uploaded, your phone should work just like it did at the beginning.

Can you copy the first few hundred lines of the serial code from Serial Monitor? (set the baud rate to 115200)

Connect the phone to the computer, select the right port, reset it (via the small button on the right hand side) and send me the text from the Serial.

**Comment by keyboard-magpie** [comment]
Very odd. Reuploaded firmware once more, left it charging and have come back to it fully working. Thanks for help.
